### ISSUE TITLE
AsyncFailsafe future(...) and futureAsync(...) methods should require SomeCallable<CompletionStage<T>> parameters

### ISSUE THREAD

**Issue by jbojar** [jhalterman/failsafe#117]
Currently `AsyncFailsafe` `future(...)` and `futureAsync(...)` methods have parameters in form `SomeKindOfCallable<CompletableFuture<T>>`. Their exact signatures are:

```
<T> java.util.concurrent.CompletableFuture<T> future(
      Callable<java.util.concurrent.CompletableFuture<T>> callable)

<T> java.util.concurrent.CompletableFuture<T> future(
      ContextualCallable<java.util.concurrent.CompletableFuture<T>> callable)

<T> java.util.concurrent.CompletableFuture<T> futureAsync(
      AsyncCallable<java.util.concurrent.CompletableFuture<T>> callable)
```

They will be more flexible if instead of `CompletableFuture` they would require `CompletionStage`. `CompletionStage` is an interface implemented by `CompletableFuture` (so `CompletableFuture` will be still automatically supported) and is sufficient for `future` and `futureAsync` methods implementations.

New signatures could be as follows:

```
<T> java.util.concurrent.CompletableFuture<T> future(
      Callable<? extends java.util.concurrent.CompletionStage<T>> callable)

<T> java.util.concurrent.CompletableFuture<T> future(
      ContextualCallable<? extends java.util.concurrent.CompletionStage<T>> callable)

<T> java.util.concurrent.CompletableFuture<T> futureAsync(
      AsyncCallable<? extends java.util.concurrent.CompletionStage<T>> callable)
```

This additional flexibility would make Failsafe library more convenient to integrate for example with Scala, where `scala.compat.java8.FutureConverters` methods convert scala futures to java `CompletionStages` (they cannot convert to `CompletableFuture` because of some possible race condition between scala `Promise.complete` and java `CompletableFuture.complete` methods).

**Comment by fabienrenaud** [comment]
All my APIs return CompletionStage. Not CompletableFuture. It would be nice to have failsafe supporting CompletionStage as as first class citizen.
Work around is to call toCompletableFuture everywhere, which works but isn't great...

**Comment by jhalterman** [comment]
Fixed by #118.
